x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorIndexNameAction.java

```diff
@@ -0,0 +1,146 @@
+
+        private final String connectorId;
+        private final String indexName;
+
+        public Request(String connectorId, String indexName) {
+            this.connectorId = connectorId;
+            this.indexName = indexName;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorId = in.readString();
+            this.indexName = in.readString();
+        }
+
+        public String getConnectorId() {
+            return connectorId;
+        }
+
+        public String getIndexName() {
+            return indexName;
+        }
+
+        private static final ConstructingObjectParser<UpdateConnectorIndexNameAction.Request, String> PARSER =
+            new ConstructingObjectParser<>(
+                "connector_update_index_name_request",
+                false,
+                ((args, connectorId) -> new UpdateConnectorIndexNameAction.Request(connectorId, (String) args[0]))
+            );
+
+        static {
+            PARSER.declareString(constructorArg(), Connector.INDEX_NAME_FIELD);
+        }
+
+        public static UpdateConnectorIndexNameAction.Request fromXContentBytes(
+            String connectorId,
+            BytesReference source,
+            XContentType xContentType
+        ) {
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, source, xContentType)) {
+                return UpdateConnectorIndexNameAction.Request.fromXContent(parser, connectorId);
+            } catch (IOException e) {
+                throw new ElasticsearchParseException("Failed to parse: " + source.utf8ToString(), e);
+            }
+        }
+
+        public static UpdateConnectorIndexNameAction.Request fromXContent(XContentParser parser, String connectorId) throws IOException {
+            return PARSER.parse(parser, connectorId);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.field(Connector.INDEX_NAME_FIELD.getPreferredName(), indexName);
+            }
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(connectorId)) {
+                validationException = addValidationError("[connector_id] cannot be [null] or [\"\"].", validationException);
+            }
+
+            if (Strings.isNullOrEmpty(indexName)) {
+                validationException = addValidationError("[index_name] cannot be [null] or [\"\"].", validationException);
+            }
+
+            try {
+                MetadataCreateIndexService.validateIndexOrAliasName(indexName, InvalidIndexNameException::new);
```

Comment:
There's also https://github.com/elastic/elasticsearch/blob/5dec83f69e808355d48161b9f1ec1f8b7a0b4b02/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java#L168-L185, which calls `validateIndexOrAliasName` to check the syntactical rules + also checks in the cluster state, if the name is already in use. Have you considered this method?

Reply:
@timgrein  I considered using this, but:
- `ResourceAlreadyExistsException` is not really useful to us, we actually expect the index to exist so that we can point the connector to it 
-`validateIndexName` it depends on `ClusterState`, it's something we have access to on transport level but not sure if we can use it for action validation, therefore I went with simpler validation

Reply:
Do you know what happens, if you update the `_index_name` to one which already exists?

Reply:
> You mean what happens if a different connector already points to an index name. Fair point, we might check this scenario in ConnectorIndexService as it would require querying for existing connectors with a given index name.

Yes, that's what I meant. I think we should avoid having different connectors point to the same index as this could have some unpleasant implications: different sources pointing to one index -> ids differ vastly -> a lot of docs get deleted on accident

Reply:
Added and tested the check! :) unfortunately can't add unit tests, the index mapping are automatically inferred (instead used from template) and therefore the term query won't work for unit tests. I don't think it's a blocking issue

Reply:
Probably the same issue I've encountered in ent-search issue #6351, sounds good to me to address it later